sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpCredentialPolicy.java

```diff
@@ -13,6 +17,17 @@ public abstract class HttpCredentialPolicy implements HttpPipelinePolicy {
     public HttpCredentialPolicy() {
     }
 
+    /**
+     * Applies the credential to the request.
+     *
+     * @param httpRequest The HTTP request to apply the credential to.
+     * @param next The next policy in the pipeline.
+     * @return The response from the next policy in the pipeline.
+     * @throws IllegalStateException If the request is not using the HTTPS scheme.
+     */
+    @Override
+    public abstract Response<BinaryData> process(HttpRequest httpRequest, HttpPipelineNextPolicy next);
```

Comment:
Is this required?

Reply:
My thought was I wanted to change the javadoc here to be specific to `HttpCredentialPolicy` instead of the base `HttpPipelinePolicy` doc. If there's a better way to accomplish that I'm happy to swap for something else, or just copy this doc to the two places that get it. (Which in effect I had to kind of do to work around `@throws` seemingly not inheriting.)

Reply:
If we wanted, we could turn this into a final method and have another abstract method all subtypes need to implement that actually adds in the credential.

This method then could be all the common validation we perform, such as the HTTPS requirement. Rather than needing all subtypes to implement it.